pkg/koordlet/metricsadvisor/collectors/performance/performance_collector_linux.go

```diff
@@ -164,25 +165,34 @@ func (p *performanceCollector) getAndStartCollectorOnSingleContainer(podParentCg
 }
 
 func (p *performanceCollector) profilePerfOnSingleContainer(containerStatus *corev1.ContainerStatus, collector *perf.PerfCollector, pod *corev1.Pod) {
+	pciMetrics := make([]metriccache.MetricSample, 0)
 	collectTime := time.Now()
 	cycles, instructions, err := util.GetContainerCyclesAndInstructions(collector)
 	if err != nil {
 		klog.Errorf("collect container %s cpi err: %v", containerStatus.Name, err)
 		return
 	}
-	containerCpiMetric := &metriccache.ContainerInterferenceMetric{
-		MetricName:  metriccache.MetricNameContainerCPI,
-		PodUID:      string(pod.UID),
-		ContainerID: containerStatus.ContainerID,
-		MetricValue: &metriccache.CPIMetric{
-			Cycles:       cycles,
-			Instructions: instructions,
-		},
+
+	pciCycle, err01 := metriccache.ContainerCPICycleMetric.GenerateSample(metriccache.MetricPropertiesFunc.PodContainer(string(pod.GetUID()), containerStatus.ContainerID), collectTime, float64(cycles))
+	pciInstruction, err02 := metriccache.ContainerCPIInstructionMetric.GenerateSample(metriccache.MetricPropertiesFunc.PodContainer(string(pod.GetUID()), containerStatus.ContainerID), collectTime, float64(instructions))
+
+	if err01 != nil || err02 != nil {
+		klog.Warningf("failed to collect Container PCI, Cycle err: %s, Instruction err: %s", err01, err02)
+		return
 	}
-	err = p.metricCache.InsertContainerInterferenceMetrics(collectTime, containerCpiMetric)
-	if err != nil {
-		klog.Errorf("insert container cpi metrics failed, err %v", err)
+
+	pciMetrics = append(pciMetrics, pciCycle, pciInstruction)
+	appender := p.metricCache.Appender()
+	if err := appender.Append(pciMetrics); err != nil {
+		klog.ErrorS(err, "Append container cycle&instruction metrics error")
+		return
 	}
+
+	if err := appender.Commit(); err != nil {
```

Comment:
how about we let profilePerfOnSingleContainer return all metric samples and do the append/commit together after the loop `for containerStatus, podMeta := range containerStatusesMap`